$OpenBSD: patch-tests_test_wsgi_py,v 1.1 2019/10/04 15:06:31 kevlo Exp $

- Fix unit tests in tests/test_wsgi.py
  upstream commit a345caedd53ea690b8804d461ae4d941b3a849b4

Index: tests/test_wsgi.py
--- tests/test_wsgi.py.orig
+++ tests/test_wsgi.py
@@ -1,9 +1,15 @@
 import unittest
-import web
 import threading
 import time
 
+try:  # PY 3
+    from urllib.parse import unquote_to_bytes as unquote
+except ImportError:  # PY 2
+    from urllib import unquote
 
+import web
+
+
 class WSGITest(unittest.TestCase):
     def test_layers_unicode(self):
         urls = ("/", "uni")
@@ -18,7 +24,7 @@ class WSGITest(unittest.TestCase):
         thread.start()
         time.sleep(0.5)
 
-        b = web.browser.Browser()
+        b = web.browser.AppBrowser(app)
         r = b.open("/").read()
         s = r.decode("utf8")
         self.assertEqual(s, u"\u0C05\u0C06")
@@ -39,7 +45,7 @@ class WSGITest(unittest.TestCase):
         thread.start()
         time.sleep(0.5)
 
-        b = web.browser.Browser()
+        b = web.browser.AppBrowser(app)
         r = b.open("/")
         self.assertEqual(r.read(), b"abcdef")
 
@@ -59,9 +65,10 @@ class WSGITest(unittest.TestCase):
         thread.start()
         time.sleep(0.5)
 
-        b = web.browser.Browser()
+        b = web.browser.AppBrowser(app)
         r = b.open("/%E2%84%A6")
-        self.assertEqual(r.read(), b"\xE2\x84\xA6")
+        s = unquote(r.read())
+        self.assertEqual(s, b"\xE2\x84\xA6")
 
         app.stop()
         thread.join()
